src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -97,9 +127,17 @@ void LambdaFormInvokers::regenerate_holder_classes(TRAPS) {
     assert(h_bytes != NULL, "Class bytes is NULL");
 
     char *class_name = java_lang_String::as_utf8_string(h_name());
+    if (class_name == nullptr) {
+      log_info(cds)("Out of memory when reloading classes, quit");
+      return;
+    }
     int len = h_bytes->length();
     // make a copy of class bytes so GC will not affect us.
     char *buf = resource_allocate_bytes(THREAD, len);
+    if (buf == nullptr) {
```

Comment:
`resource_allocate_bytes(THREAD, len)` by default uses `AllocFailStrategy::EXIT_OOM`. We should explicitly pass `RETURN_NULL` instead.

Also, the `resource_free_bytes` call below doesn't seem necessary. The buffer will be freed when the ResourceMark is popped.

Reply:
Done.